### PR TITLE
Add new endpoint for fetching rule evaluations

### PR DESCRIPTION
**Pull request by evankanderson**
# Summary

Defines a new API for fetching rule evaluation results, rather than overloading `GetProfileStatusByName`.  Does not provide an implementation, though hopefully the implementation is straightforward.

## Change Type

***Mark the type of change your PR introduces:***

- [ ] Bug fix (resolves an issue without affecting existing features)
- [x] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [ ] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

```
make gen
make test
make build
```

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [x] Added comments to complex or tricky code sections.
- [x] Updated any affected documentation.
- [ ] Included tests that validate the fix or feature.
- [x] Checked that related changes are merged.

### CODE REVIEW

**Review by dmjb — DISMISSED** [review]

File: proto/minder/v1/minder.proto [hunk 1]
```diff
@@ -1128,6 +1149,45 @@ message DeleteRuleTypeResponse {
...
+        // Filter profiles to only those matching the specified labels.
+        // 
+        // The default is to return all user-created profiles; the string "*" can
+        // be used to select all profiles, including system profiles.  This syntax
+        // may be expanded in the future.
+        string label_filter = 3;
+    }
+
+    // If set, only return evaluation results for the named rules.
+    // If empty, return evaluation results for all rules
```
Comment (rdimitrov) [inline proto/minder/v1/minder.proto:1173]: I guess it's not intentional but the comment here is the same as the next one

File: proto/minder/v1/minder.proto [hunk 2]
```diff
@@ -1438,6 +1498,11 @@ message Profile {
     // name is the name of the profile instance.
     string name = 3;
 
+    // labels are a set of system-provided attributes which can be used to
+    // filter profiles and status results.  Labels cannot be set by the user,
+    // but are returned in ListProfiles.
+    repeated string labels = 12;
```
Comment (JAORMX) [inline proto/minder/v1/minder.proto:1504]: Are we gonna have labels be a fixed list of strings? or are we going to have labels be a keypair like k8s? This is something I'd like to get consensus on.

**Review by eleftherias — COMMENTED** [review]
Is this in preparation for label-based profiles? Or something else?

**Comment by JAORMX** [comment]
> Is this in preparation for label-based profiles? Or something else?

yep

**Review by evankanderson — COMMENTED** [review]
> Are we gonna have labels be a fixed list of strings? or are we going to have labels be a keypair like k8s? This is something I'd like to get consensus on. 

Yeah, that's exactl the sort of thing I wanted to highlight with this API proposal.

I was thinking bag-of-strings, but we could do keypairs if that seems more useful.

**Comment by JAORMX** [comment]
@evankanderson thinking about this further, I don't particularly have a use-case for key-value style labels. So, an array of strings is just as good in this case.

**Comment by evankanderson** [comment]
My first thought was to start with dns-style labels, but I'm wondering if that's too constricting.

**Comment by JAORMX** [comment]
> My first thought was to start with dns-style labels, but I'm wondering if that's too constricting.

I'd say DNS + a minimal subset of characters like "/" and "-" should be fine to start with

**Review by rdimitrov — APPROVED** [review]

File: proto/minder/v1/minder.proto (hunk 2 above)
Comment (evankanderson) [reply proto/minder/v1/minder.proto:1504]: Added details on the allowed strings.

File: proto/minder/v1/minder.proto (hunk 1 above)
Comment (evankanderson) [reply proto/minder/v1/minder.proto:1173]: Oops, thanks, fixed!

**Comment by evankanderson** [comment]
Okay, I updated the comments based on our discussion about K:V maps vs "bag of strings".  We ended up going with the bag of strings approach because all our "values" ended up being singletons (e.g. "true" or "hidden", but no other value).  Also clarified max size and allowed characters for the labels.

File: proto/minder/v1/minder.proto (hunk 2 above)
Comment (JAORMX) [reply proto/minder/v1/minder.proto:1513]: For the record, we've agreed that an array of labels was the way to go and we'd have a DNS-style restricted character set to begin with. Thanks @evankanderson for driving this!

**Review by JAORMX — APPROVED** [review]

**Pull request merged**